File: sample/src/main/java/com/schibsted/spain/barista/sample/ViewPagerWithTwoDifferentPagesActivity.java
package com.schibsted.spain.barista.sample;

import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;
import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;

public class ViewPagerWithTwoDifferentPagesActivity extends FragmentActivity {

  private static final int NUM_PAGES = 2;

  private ViewPager pager;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_viewpager);

    PagerAdapter adapter = new ScreenSlidePagerAdapter(getSupportFragmentManager());
    pager = (ViewPager) findViewById(R.id.pager);
    pager.setAdapter(adapter);
  }

  @Override
  public void onBackPressed() {
    if (pager.getCurrentItem() == 0) {
      super.onBackPressed();
    } else {
      pager.setCurrentItem(pager.getCurrentItem() - 1);
    }
  }

  private static class ScreenSlidePagerAdapter extends FragmentStatePagerAdapter {
    ScreenSlidePagerAdapter(FragmentManager fm) {
      super(fm);
    }

    @Override
    public Fragment getItem(int position) {
      if (position == 0) {
        return new ViewPagerFirstFragment();
      } else {
        return new ViewPagerSecondFragment();
      }
    }

    @Override
    public int getCount() {
      return NUM_PAGES;
    }
  }
}

File: sample/src/main/java/com/schibsted/spain/barista/sample/ViewPagerWithTwoDifferentPagesActivity.kt
package com.schibsted.spain.barista.sample

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager

class ViewPagerWithTwoDifferentPagesActivity : FragmentActivity() {
  private var pager: ViewPager? = null
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_viewpager)
    val adapter: PagerAdapter = ScreenSlidePagerAdapter(
      supportFragmentManager)
    pager = findViewById<View>(R.id.pager) as ViewPager
    pager!!.adapter = adapter
  }

  override fun onBackPressed() {
    if (pager!!.currentItem == 0) {
      super.onBackPressed()
    } else {
      pager!!.currentItem = pager!!.currentItem - 1
    }
  }

  private class ScreenSlidePagerAdapter internal constructor(fm: FragmentManager?) : FragmentStatePagerAdapter(
    fm!!) {
    override fun getItem(position: Int): Fragment {
      return if (position == 0) {
        ViewPagerFirstFragment()
      } else {
        ViewPagerSecondFragment()
      }
    }

    override fun getCount(): Int {
      return NUM_PAGES
    }
  }

  companion object {
    private const val NUM_PAGES = 2
  }
}